### PR TITLE
Complete Rfc 8037 support

### LINKED ISSUE #127: Implement support for CFRG Curves (Ed/X-25519/448)
RFC 8037 defines how to use CRFG curves Ed25519 and Ed448 for EdDSA digital signatures and X25519 and X448 for ECDH key exchange.

### PR DESCRIPTION
**Pull request by simo5**
Adds X25519 and X448 ECDH-ES exchange support

Fixes #127

### CODE REVIEW

**Comment by simo5** [comment]
@tiran, PTAL

File: jwcrypto/jwk.py [hunk 1]
```diff
@@ -753,11 +819,11 @@ def import_from_pyca(self, key):
             self._import_pyca_pri_ec(key)
         elif isinstance(key, ec.EllipticCurvePublicKey):
             self._import_pyca_pub_ec(key)
-        elif isinstance(key, ed25519.Ed25519PrivateKey) or \
-                isinstance(key, ed448.Ed448PrivateKey):
+        elif isinstance(key, Ed25519PrivateKey) or \
```
Comment (tiran) [inline jwcrypto/jwk.py]: isinstance supports tuple of types: ``isinstance(key, (Ed25519PrivateKey, ed448.Ed448PrivateKey)):``

File: jwcrypto/jwk.py [hunk 2]
```diff
@@ -753,11 +819,11 @@ def import_from_pyca(self, key):
...
+                isinstance(key, Ed448PrivateKey):
             self._import_pyca_pri_okp(key)
-        elif isinstance(key, ed25519.Ed25519PublicKey) or \
-                isinstance(key, ed448.Ed448PublicKey):
+        elif isinstance(key, Ed25519PublicKey) or \
```
Comment (tiran) [inline jwcrypto/jwk.py]: ``isinstance(key, (Ed448PublicKey, Ed25519PublicKey))``

File: jwcrypto/jwk.py [hunk 3]
```diff
@@ -367,22 +424,19 @@ def _import_pyca_pub_ec(self, key, **params):
...
-            crv = params['crv']
-        # Make crv available in params for _import_pyca_*_okp() methods
-        params['crv'] = crv
-        if crv == 'Ed25519':
-            key = ed25519.Ed25519PrivateKey.generate()
-        elif crv == 'Ed448':
-            key = ed448.Ed448PrivateKey.generate()
+        if 'crv' not in params:
+            raise InvalidJWKValue('Must specify "crv" for OKP key generation')
+        if params['crv'] == 'Ed25519':
```
Comment (tiran) [inline jwcrypto/jwk.py]: This ``if/elif/elif`` approach is hard to maintain. How about an internal mapping from name to type?

```
_CURVE_CLASSES = {
   'Ed25519': (Ed25519PublicKey, Ed25519PrivateKey),
   ...
}
```

Then you can replace ``name in ['Ed25519', 'Ed448', 'X25519', 'X448']`` with ``name in _CURVE_CLASSES`` and the lookup with ``pubclass, privclass = _CURVE_CLASSES.get(params['crv'], (None, None))`` and a check like ``if pubclass is not None: return pubclass.generate()``.

File: jwcrypto/jwk.py (hunk 1 above)
Comment (simo5) [reply jwcrypto/jwk.py]: ok

File: jwcrypto/jwk.py (hunk 2 above)
Comment (simo5) [reply jwcrypto/jwk.py]: ok

File: jwcrypto/jwk.py (hunk 3 above)
Comment (simo5) [reply jwcrypto/jwk.py]: Yeah didn't like the if/then/else myself, I will try to implement your suggestion and see if it looks better as it sounds promising.

Comment (simo5) [reply jwcrypto/jwk.py]: Please check latest rebase, I added a patch to use a lookup table everywhere possible

**Pull request merged**